o1vm/run-vm.sh

```diff
@@ -4,40 +4,48 @@ set -euo pipefail
 O1VM_FLAVOR="${O1VM_FLAVOR:-pickles}"
 
 case ${O1VM_FLAVOR} in
-    "legacy")
-        BINARY_FLAVOR="legacy_o1vm"
-        ;;
-    "pickles")
-        BINARY_FLAVOR="pickles_o1vm"
-        ;;
-    *)
-        echo "${BASH_SOURCE[0]}:${LINENO}: only flavors 'legacy' and \
+"legacy")
+    BINARY_FLAVOR="legacy_o1vm"
+    ;;
+"pickles")
+    BINARY_FLAVOR="pickles_o1vm"
+    ;;
+*)
+    echo "${BASH_SOURCE[0]}:${LINENO}: only flavors 'legacy' and \
 'pickles' (default) are supported for now. Use the environment variable \
-O1VM_FLAVOR to one of these values to run the flavor you would like";
-        exit 1
-        ;;
-    esac
+O1VM_FLAVOR to one of these values to run the flavor you would like"
+    exit 1
+    ;;
+esac
 
-cargo run --bin ${BINARY_FLAVOR} \
+if [ -d "op-program-db-for-latest-l2-block" ] && [ -f "snapshot-state-3000000.json" ]; then
```

Comment:
Same as above. I would simply change how the env file is generated in the CI (?) to keep it simple.